### PR TITLE
Adds better logging around SMB mapping for FSX

### PR DESCRIPTION
**Pull request by jterry75**
### Summary
1. Adds debug logging for SMB mapping for the command
that is being executed. This command will not not log the
creds used to attach the mapping.

2. Logs the ps output stderr/stdout from the command on error.

Signed-off-by: Justin Terry <jlterry@amazon.com>

### Implementation details
Added the logging around the PowerShell exec.

### Testing
Have verified that in both the debug cmdline and the output it doesnt log the password.

### Description for the changelog
Adds command and error logging for FSx file mapping when calling out to PowerShell.

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jterry75** [comment]
@vsiddharth - Can you give me some feedback on this.

File: agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go [hunk 1]
```diff
@@ -552,18 +552,25 @@ func (fv *FSxWindowsFileServerResource) performHostMount(remotePath string, user
...
 		remotePathArg,
 		creds,
 		"-Persistent $true",
 		"-RequirePrivacy $true",
-		"-ErrorAction Stop")
+		"-ErrorAction Stop",
+	}
+	log_args := make([]string, len(args))
+	log_args = append(log_args, args...)
+	log_args[3] = "<creds>" // Clear the creds for logging.
```
Comment (vsiddharth) [inline agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go:567]: I think the debug log below would include cred like below
```
-Credential $(New-Object System.Management.Automation.PSCredential(\"...", $(ConvertTo-SecureString \"...\" -AsPlainText -Force)))
```

File: agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go [hunk 2]
```diff
@@ -552,18 +552,25 @@ func (fv *FSxWindowsFileServerResource) performHostMount(remotePath string, user
...
+	seelog.Debugf("Executing mapping of fsxwindowsfileserver with cmd: %v %v", strings.Join(log_args[:3], " "), strings.Join(log_args[4:], " "))
 
-	_, err = cmd.CombinedOutput()
+	cmd := execCommand("powershell.exe", args...)
+	out, err := cmd.CombinedOutput()
 	if err != nil {
-		seelog.Errorf("Failed to map fsxwindowsfileserver resource on the container instance: %v", err)
+		seelog.Errorf("Failed to map fsxwindowsfileserver resource on the container instance error: %v, out: %v", err, string(out))
```
Comment (vsiddharth) [inline agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go:573]: Thanks a lot for this change. This should be very helpful for customers.

File: agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go (hunk 1 above)
Comment (jterry75) [reply agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go]: Hey I removed the extra copy. Did you mean the error log shows it still? The index is skipped in the debug log so it wouldnt have it.

File: agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go (hunk 2 above)
Comment (jterry75) [reply agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go:571]: Np

**Review by vsiddharth — APPROVED** [review]
LGTM :shipit: 

Thanks a lot for making these changes.

File: agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go (hunk 2 above)
Comment (singholt) [reply agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go:571]: does `out` print creds if its incorrect? just making sure

Comment (jterry75) [reply agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go]: It does not print them in "out" but I added a replace just to be extra cautious incase PS ever changes behavior here.

**Comment by jterry75** [comment]
New output:
```
level=debug time=2022-06-09T20:08:24Z msg="Executing mapping of fsxwindowsfileserver with cmd: New-SmbGlobalMapping -LocalPath \"Z:\" -RemotePath \"\\\\amznfsxricml6z3.sidvin.fsx.com\\share\" -Persistent $true -RequirePrivacy $true -ErrorAction Stop" module=fsxwindowsfileserver_windows.go
level=error time=2022-06-09T20:08:26Z msg="Failed to map fsxwindowsfileserver resource on the container instance error: exit status 1, out: New-SmbGlobalMapping : The password of this user has expired. \r\nAt line:1 char:1\r\n+ New-SmbGlobalMapping -LocalPath \"Z:\" -RemotePath \"\\\\amznfsxricml6z3.s ...\r\n+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\r\n    + CategoryInfo          : NotSpecified: (MSFT_SmbGlobalMapping:ROOT/Microsoft/...mbGlobalMapping) [New-SmbGlobalMa \r\n   pping], CimException\r\n    + FullyQualifiedErrorId : Windows System Error 2242,New-SmbGlobalMapping\r\n \r\n" module=fsxwindowsfileserver_windows.go
```

File: agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go [hunk 3]
```diff
@@ -553,18 +553,23 @@ func (fv *FSxWindowsFileServerResource) performHostMount(remotePath string, user
...
+		"-ErrorAction Stop",
+	}
+	seelog.Debugf("Executing mapping of fsxwindowsfileserver with cmd: %v %v", strings.Join(args[:3], " "), strings.Join(args[4:], " "))
 
-	_, err = cmd.CombinedOutput()
+	cmd := execCommand("powershell.exe", args...)
+	out, err := cmd.CombinedOutput()
 	if err != nil {
-		seelog.Errorf("Failed to map fsxwindowsfileserver resource on the container instance: %v", err)
+		safeOutput := strings.ReplaceAll(string(out), password, "<pass>")
```
Comment (singholt) [inline agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go:571]: thanks! wouldnt this just look for "password" in `out` and replace it and not the actual password itself? 🤔

Comment (jterry75) [reply agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go:571]: `password` is a parameter from above not a string. Right?

Comment (singholt) [reply agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go:571]: ah my bad

**Review by singholt — APPROVED** [review]

File: agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go (hunk 3 above)
Comment (jterry75) [reply agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go:571]: NP

**Review by fierlion — APPROVED** [review]

**Pull request merged**